### ISSUE TITLE
Don't lazy load handler group routes.

### ISSUE THREAD

**Issue by lee-sony** [gogf/gf#4039]
### Is your feature request related to a problem?

Option Yes

### Describe the solution you'd like

#682
As described in the above issue, RouterGroup routes are lazy loaded. This is this different behaviour from Go's native http mux. This behaviours makes isolated unit tests impossible.

### Describe alternatives you've considered

Using it with `g.Server().Run()` but this requires ports to be open. This is not good for parallelizing a large number of unit tests.

**Comment by gqcn** [comment]
Your question is very in-depth and professional! In fact, the lazy initialization of grouped routing is only part of the Server initialization. The complete lazy initialization logic is here: https://github.com/gogf/gf/blob/0c2d5cac193498e3ecb91b11d4f24a136a0a7283/net/ghttp/ghttp_server.go#L124 .

I am also considering how to efficiently unit test the Server routing. Maybe it would be a good idea for us to add an initialization method that doesn't actually listen on the port. For example, adding a new `Server.Init` method to perform some initialization operations. 

What do you think?

**Comment by ninjashixuan** [comment]
I encountered a similar issue, though not specifically with httptest. Instead, it arose when I was using goai to generate multi-version API documentation (e.g., v1, v2, v3). The problem lies in the lazy loading behavior of s.GetRoutes. Since routes are only available after s.Run(), I had to use a workaround: waiting for s.Run() in another goroutine, then calling GetRoutes, and finally using BindHandler to set up the API versions.

While this approach works, it feels inelegant and adds unnecessary complexity.

**Comment by gqcn** [comment]
> I encountered a similar issue, though not specifically with httptest. Instead, it arose when I was using goai to generate multi-version API documentation (e.g., v1, v2, v3). The problem lies in the lazy loading behavior of s.GetRoutes. Since routes are only available after s.Run(), I had to use a workaround: waiting for s.Run() in another goroutine, then calling GetRoutes, and finally using BindHandler to set up the API versions.
> 
> While this approach works, it feels inelegant and adds unnecessary complexity.

I agree! So is it a good idea to adding a `Server.Init` method that does routes initialization but no port listening?

**Comment by ninjashixuan** [comment]
> > I encountered a similar issue, though not specifically with httptest. Instead, it arose when I was using goai to generate multi-version API documentation (e.g., v1, v2, v3). The problem lies in the lazy loading behavior of s.GetRoutes. Since routes are only available after s.Run(), I had to use a workaround: waiting for s.Run() in another goroutine, then calling GetRoutes, and finally using BindHandler to set up the API versions.
> > While this approach works, it feels inelegant and adds unnecessary complexity.
> 
> I agree! So is it a good idea to adding a `Server.Init` method that does routes initialization but no port listening?

I think it's feasible.

**Comment by kylelmh** [comment]
Maybe call it `Server.PreloadRoutes` or something to prevent confusion, and then add some documentation about unit tests on the site.

**Comment by ninjashixuan** [comment]
一年多的回访，@gqcn 这个feature 怎么还没实现。
